components/dashboard/src/teams/TeamUsage.tsx

```diff
@@ -64,8 +75,20 @@ function TeamUsage() {
                 to: endDateOfBillMonth,
             };
             try {
-                const billedUsageResult = await getGitpodService().server.listBilledUsage(request);
-                setBilledUsage(billedUsageResult);
+                const { server } = getGitpodService();
+                const billedUsageResult = await server.listBilledUsage(request);
+                const extendedResult = await Promise.all(
+                    billedUsageResult.map(async (res) => {
+                        const ws = await server.getWorkspaceForBillableSession(res.workspaceId, attributionId);
+                        if (!res.userId) {
```

Comment:
Comments:
- I decided on making the extra calls here instead of extending the `listBilledUsage` in the server because `listBilledUsage` is currently being called somewhere else as well (when we check the spending limit). That call will be replaced eventually, though. Maybe we can move the calls then, but go with this in this iteration?
- Happy to get a review on how to write this convoluted block better 🙏🏽

Reply:
@laushinka Now I see that we meant with the additional calls. I don't like that they introduce extra API, and really thing the perfect spot for it would be [here](https://github.com/gitpod-io/gitpod/blob/b4a9939063fad70d25474e5a76f0b5923e27e43e/components/server/ee/src/workspace/gitpod-server-impl.ts#L2175).WDYT?
Biggest benefit I see is that we would resolve the 1+N situation, and can try to replace it with a single SQL if we do it right.

Reply:
Agree there would be best - only did this as a temporary solution as I mentioned above because the spending limit also calls `listBilledUsage` for now. 
Will make the change.